### PR TITLE
Return empty WorkflowContextList from GetWorkflowContextList

### LINKED ISSUE #391: Listing workflows by hardware without any workflows returns GRPC error
## Expected Behaviour
When listing workflows by hardware, which do not have any workflows assigned, empty list of workflows should be returned.

## Current Behaviour
Currently server logs the following error and returns error to the client, when given hardware ID does not have any associated workflows:
```
tink-server_1  | ERROR: 2020/12/14 13:52:48 grpc: server failed to encode response: rpc error: code = Internal desc = grpc: error while marshaling: proto: Marshal called with nil
```

```
    main_test.go:38: listing workflows should succeed, got: rpc error: code = Internal desc = grpc: error while marshaling: proto: Marshal called with nil
```

## Steps to Reproduce (for bugs)

1. `go mod init foo`
2. Create file called `main_test.go` with the following content:
```go
package main

import (
  "context"
  "os"
  "testing"

  "github.com/google/uuid"
  "github.com/tinkerbell/tink/client"
  "github.com/tinkerbell/tink/protos/workflow"
)

func listWorkflowsByHardwareID(t *testing.T, hardwareID string) *workflow.WorkflowContextList {
  if hardwareID == "" {
    t.Fatalf("Hardware ID can't be empty")
  }

  envVars := map[string]string{
    "TINKERBELL_GRPC_AUTHORITY": "127.0.0.1:42113",
    "TINKERBELL_CERT_URL":       "http://127.0.0.1:42114/cert",
  }

  for k, v := range envVars {
    if err := os.Setenv(k, v); err != nil {
      t.Fatalf("setting %q environment variable: %v", k, err)
    }
  }

  conn, err := client.GetConnection()
  if err != nil {
    t.Fatalf("getting Tinkerbell connection: %v", err)
  }

  c := workflow.NewWorkflowSvcClient(conn)

  list, err := c.GetWorkflowContextList(context.TODO(), &workflow.WorkflowContextRequest{WorkerId: hardwareID})
  if err != nil {
    t.Fatalf("listing workflows should succeed, got: %v", err)
  }

  return list
}

func Test_Listing_workflows_by_non_existent_hardware_ID_does_return_empty_list(t *testing.T) {
  listWorkflowsByHardwareID(t, uuid.New().String())
}

func Test_Listing_workflows_for_existing_hardware_ID_does_return_empty_list(t *testing.T) {
  l := listWorkflowsByHardwareID(t, "9ef087c3-4d9f-16ed-777a-ae7583fc5030")

  if len(l.GetWorkflowContexts()) == 0 {
    t.Fatalf("list should not be empty")
  }
}
```
3. Make sure values for env variables and hardware ID in `Test_Listing_workflows_for_existing_hardware_ID_does_return_empty_list` test are set correctly.
4. Run `go test -v`.

## Context
While working on https://github.com/tinkerbell/cluster-api-provider-tink, we want to find hardware IDs which do not have any workflows associated, to select them
as "available" workers.

## Your Environment

Tinkerbell version 9c52cca5.

### PR DESCRIPTION
**Pull request by swills**
Fixes #391

### CODE REVIEW

**Comment by swills** [comment]
Needs a test...

File: grpc-server/tinkerbell.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func (s *server) GetWorkflowContextList(ctx context.Context, req *pb.WorkflowCon
 			WorkflowContexts: wfContexts,
 		}, nil
 	}
-	return nil, nil
+	return &pb.WorkflowContextList{WorkflowContexts: []*pb.WorkflowContext{}}, nil
```
Comment (displague) [inline grpc-server/tinkerbell.go:71]: Can we avoid this just by removing the `nil` check on line 58? As long as `wfs` is a slice type, the empty `range` loop should have the same result.

Comment (displague) [reply grpc-server/tinkerbell.go:71]: (remove line 58, 70, 71 .. done?)

Comment (swills) [reply grpc-server/tinkerbell.go]: Did that, thanks for the suggestion.

**Comment by displague** [comment]
@swills See the DCO notes https://github.com/tinkerbell/tink/pull/575/checks?check_run_id=4695236298 and please squash the second commit into the first.

I tried to reproduce the `make generate` problem locally but I ran into problems with `make tools` not being providing the tools needed for `make generate`.

**Comment by swills** [comment]
> @swills See the DCO notes https://github.com/tinkerbell/tink/pull/575/checks?check_run_id=4695236298 and please squash the second commit into the first.
> 

Done

> I tried to reproduce the `make generate` problem locally but I ran into problems with `make tools` not being providing the tools needed for `make generate`. For CI, this may be corrected by [73ca7c7](https://github.com/tinkerbell/tink/commit/73ca7c792f2715d02b111c35377a6f47388abb7d)

Maybe. The real fix imho is https://github.com/actions/setup-go/pull/175 see also https://github.com/tinkerbell/tink/pull/553

**Comment by jacobweinstock** [comment]
Hey @swills, sorry for having this sit so long. Would you mind checking out the merge/rebase conflicts?

**Comment by chrisdoherty4** [comment]
Thanks for the effort. Unfortunately this change is redundant because of https://github.com/tinkerbell/tink/pull/654.

**Pull request closed without merging**